sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/DoNotInstantiateSharedClasses.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System.ComponentModel.Composition;
+
+namespace Tests.Diagnostics
+{
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    class SharedClass {}
+
+    [PartCreationPolicy(CreationPolicy.NonShared)]
+    class NonSharedClass { }
+
+    [PartCreationPolicy(CreationPolicy.Any)]
+    class AnyClass { }
+
```

Comment:
Please add a test with the full namespace.

Reply:
Please add a test with no ctor param (e.g. `[PartCreationPolicy()]`).

Reply:
Great suggestions!